backend/contracts/BetOnChain.sol

```diff
@@ -0,0 +1,108 @@
+//SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.18;
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {BocNFT} from './BocNFT.sol';
+import {BocToken} from './BocToken.sol';
+
+error BetOnChain__RequirementsNotMet();
+
+contract BetOnChain is Ownable {
```

Comment:
This is just a comment, we should do this on a separate ticket.
 
 
This is a good start, I think we can base it a lot more on the Lottery contract, with start/end times for bets, possibility of winner(s) to withdraw the available prizes. In addition it should be a contract factory, like in https://blog.logrocket.com/cloning-solidity-smart-contracts-factory-pattern/.